src/main/java/edu/wpi/grip/core/SocketHint.java

```diff
@@ -56,6 +69,14 @@ public SocketHint(String identifier, Class<T> type, T initialValue) {
         this(identifier, type, () -> initialValue);
     }
 
+    protected void setPublishable(boolean publishable) {
```

Comment:
I don't see the value in this function. If you want to make a socket publishable then it should be set in the constructor.